File: 0001_0100/0050/0050.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = f.readline().strip()
    t = f.readline().strip()
    d = {t[i:] + t[:i] for i in range(len(t))}
    k = 0
    for x in d:
        n = s.find(x, 0)
        while n > -1:
            k += 1
            n = s.find(x, n + 1)
    q.write(str(k))

        

File: 0001_0100/0050/0050.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var s, t string
	fmt.Fscan(f, &s, &t)
	ls, lt := len(s), len(t)
	t = strings.Repeat(t, 2)
	n := 0
	for i := 0; i < ls-lt+1; i++ {
		for j := 0; j < lt; j++ {
			if s[i:i+lt] == t[j:j+lt] {
				n++
				break
			}
		}
	}
	fmt.Fprint(q, n)
}
